app/javascript/packs/components/Form.js

```diff
@@ -141,6 +160,29 @@ class Form extends React.Component{
     this.setState({ hasConfirmedOperation: true, selectedProject: "" })
   }
 
+  areValidWorkPeriods(from1, to1, from2, to2) {
+    if (compareHours({ firstHour: from1, secondHour: to1 }) && compareHours({ firstHour: from2, secondHour: to2 }))
+      return true;
+    if (compareHours({ firstHour: from1, secondHour: from2 }) && compareHours({ firstHour: to1, secondHour: to2 }))
+      return true;
```

Comment:
Could it be more clear making something like is shown below?

```suggestion
    const cond1 = compareHours({ firstHour: from1, secondHour: to1 }) && compareHours({ firstHour: from2, secondHour: to2 });
   const cond2 = compareHours({ firstHour: from1, secondHour: from2 }) && compareHours({ firstHour: to1, secondHour: to2 });
   return cond1 || cond2;
```

NOTE: You can define more meaningful names instead of `cond1` and `cond2`.

Reply:
Very nice tip @abarbosa-godaddy!!! I will improve this. Thanks